qiita_db/data.py

```diff
@@ -288,29 +288,48 @@ class RawData(BaseData):
     _table = "raw_data"
     _data_filepath_table = "raw_filepath"
     _data_filepath_column = "raw_data_id"
-    # Define here the class name, so in case it changes in the database we
-    # only need to change it here
-    _study_raw_table = "study_raw_data"
 
     @classmethod
-    def create(cls, filetype, studies, filepaths=None):
+    def create(cls, filetype, prep_templates, filepaths):
         r"""Creates a new object with a new id on the storage system
 
         Parameters
         ----------
         filetype : int
             The filetype identifier
-        studies : list of Study
-            The list of Study objects to which the raw data belongs to
-        filepaths : iterable of tuples (str, int), optional
+        prep_templates : list of PrepTemplates
+            The list of PrepTemplate objects to which the raw data is attached
+        filepaths : iterable of tuples (str, int)
             The list of paths to the raw files and its filepath type identifier
 
         Returns
         -------
         A new instance of `cls` to access to the RawData stored in the DB
+
+        Raises
+        ------
+        IncompetentQiitaDeveloperError
+            If any of the passed prep templates already have a raw data id
         """
-        # Add the raw data to the database, and get the raw data id back
         conn_handler = SQLConnectionHandler()
+        # We first need to check if the passed prep templates doesn't have
+        # a raw data already attached to them
+        sql = """SELECT EXISTS(
+                    SELECT *
+                    FROM qiita.prep_template
+                    WHERE prep_template_id IN ({})
+                        AND raw_data_id IS NOT NULL)""".format(
+            ', '.join(['%s'] * len(prep_templates)))
+        exists = conn_handler.execute_fetchone(
+            sql, [pt.id for pt in prep_templates])[0]
+        if exists:
+            raise IncompetentQiitaDeveloperError(
```

Comment:
Just checking, is it really an IncompetentQiitaDeveloperError or a QiitaError? I'm thinking than a user can hit this error from IPython.